src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILInfo.cs

```diff
@@ -101,23 +100,24 @@ public int GetTokenFor(RuntimeFieldHandle field, RuntimeTypeHandle contextType)
             throw new NotImplementedException();
         }
 
-        public void SetCode(byte[] code, int maxStackSize)
+        public void SetCode(byte[]? code, int maxStackSize)
         {
-            if (code == null)
-                throw new ArgumentNullException(nameof(code));
             method.GetILGenerator().SetCode(code, maxStackSize);
         }
 
         [CLSCompliantAttribute(false)]
         public unsafe void SetCode(byte* code, int codeSize, int maxStackSize)
         {
-            if (code == null)
+            if (codeSize < 0)
+                throw new ArgumentOutOfRangeException(nameof(codeSize), SR.ArgumentOutOfRange_GenericPositive);
+            if (codeSize > 0 && code == null)
```

Comment:
FYI, I deleted the previous argument validation because it was inconsistent with coreclr's (e.g. allowing a null code as long as codeSize was 0), and it was missing some validation.  This is a copy of what coreclr has.